arch/sim/src/sim/sim_video.c

```diff
@@ -0,0 +1,354 @@
+}
+
+static int sim_video_start_capture(imgsensor_stream_type_t type,
+                                uint8_t nr_fmt,
+                                FAR imgsensor_format_t *fmt,
+                                FAR imgsensor_interval_t *interval)
+{
+  return video_host_set_fmt(fmt[IMGDATA_FMT_MAIN].width,
+    fmt[IMGDATA_FMT_MAIN].height,
+    imgsensor_fmt_to_v4l2(fmt[IMGDATA_FMT_MAIN].pixelformat),
+    interval->denominator, interval->numerator);
+}
+
+static int sim_video_stop_capture(imgsensor_stream_type_t type)
+{
+  return 0;
+}
+
+static int sim_video_data_init()
+{
+  memset(&priv, 0, sizeof(priv));
+  return video_host_data_init();
+}
+
+static int sim_video_data_uninit()
+{
+  return 0;
+}
+
+static int sim_video_data_validate_frame_setting
+             (uint8_t nr_datafmt,
+              imgdata_format_t *datafmt,
+              imgdata_interval_t *interval)
+{
+  return 0;
+}
+
+static int sim_video_data_start_capture
+             (uint8_t nr_datafmt,
+              imgdata_format_t *datafmt,
+              imgdata_interval_t *interval,
+              imgdata_capture_t callback)
+{
+  int ret;
+
+  priv.capture_cb = callback;
+  ret = video_host_start_capture();
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  priv.is_streaming = true;
+  return 0;
+}
+
+static int sim_video_data_stop_capture()
+{
+  priv.is_streaming = false;
+  return video_host_stop_capture();
+}
+
+static int sim_video_data_validate_buf(uint8_t *addr, uint32_t size)
+{
+  if (!addr || (uintptr_t)(addr) & 0x1f)
+    {
+      return -EINVAL;
+    }
+
+  return 0;
+}
+
+static int sim_video_data_set_buf(uint8_t *addr, uint32_t size)
+{
+  priv.next_buf = (uintptr_t)addr;
+  priv.buf_size = size;
+  return 0;
+}
+
+static uint32_t imgsensor_fmt_to_v4l2(uint32_t pixelformat)
+{
+  uint32_t fourcc;
+  switch (pixelformat)
+    {
+      case IMGSENSOR_PIX_FMT_YUV420P:
+        fourcc = V4L2_PIX_FMT_YUV420;
+        break;
+
+      case IMGSENSOR_PIX_FMT_YUYV:
+        fourcc = V4L2_PIX_FMT_YUYV;
+        break;
+
+      case IMGSENSOR_PIX_FMT_JPEG_WITH_SUBIMG:
+        fourcc = V4L2_PIX_FMT_JPEG;
+        break;
+
+      case IMGSENSOR_PIX_FMT_JPEG:
+        fourcc = V4L2_PIX_FMT_JPEG;
+        break;
+
+      case IMGSENSOR_PIX_FMT_RGB565:
+        fourcc = V4L2_PIX_FMT_RGB565;
+        break;
+
+      case IMGSENSOR_PIX_FMT_UYVY:
+        fourcc = V4L2_PIX_FMT_UYVY;
+        break;
+
+      default:
+
+      /* Unsupported format */
+
+        fourcc = 0;
+    }
+
+  return fourcc;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int sim_video_initialize(void)
+{
+  imgsensor_register(&g_sim_video_ops);
+  imgdata_register(&g_sim_video_data_ops);
+  return 0;
+}
+
+int sim_video_uninitialize(void)
+{
+  return 0;
+}
+
+void sim_video_loop(void)
+{
+  int ret;
+
+  if (priv.is_streaming && priv.next_buf)
```

Comment:
remove is_streaming, it's enough to check next_buf

Reply:
done